bundles/org.openhab.binding.enocean/src/main/java/org/openhab/binding/enocean/internal/eep/D2_06/D2_06_01.java

```diff
@@ -0,0 +1,198 @@
+ * supported by a device it will transmit a 'not supported' message which is ignored by this implementation.
+ * Consequently channels that are not supported by the physical device will never send updates to linked items.
+ * 
+ * @author Thomas Lauterbach - Initial contribution
+ */
+public class D2_06_01 extends _VLDMessage {
+
+    public D2_06_01() {
+        super();
+    }
+
+    public D2_06_01(ERP1Message packet) {
+        super(packet);
+    }
+
+    protected State getWindowSashState() {
+        int sashState = bytes[2] & 0x0F;
+        switch (sashState) {
+            case 0x01:
+                return new StringType("NOT TILTED");
+            case 0x02:
+                return new StringType("TILTED");
+        }
+
+        return UnDefType.UNDEF;
+    }
+
+    protected State getWindowHandleState() {
+        int handleState = bytes[2] >>> 4;
+        switch (handleState) {
+            case 0x01:
+                return new StringType("UP");
+            case 0x02:
+                return new StringType("DOWN");
+            case 0x03:
+                return new StringType("LEFT");
+            case 0x04:
+                return new StringType("RIGHT");
+        }
+
+        return UnDefType.UNDEF;
+    }
+
+    protected State getMotionState() {
+        int motionState = bytes[4] >>> 4;
+        switch (motionState) {
+            case 0x00:
+                return OnOffType.OFF;
+            case 0x01:
+                return OnOffType.ON;
+        }
+        return UnDefType.UNDEF;
+    }
+
+    protected State getTemperature() {
+        double unscaledTemp = (double) (bytes[5] & 0xFF);
+        if (unscaledTemp <= 250) {
+            double scaledTemp = unscaledTemp * 0.32 - 20;
+            return new QuantityType<>(scaledTemp, SIUnits.CELSIUS);
+        }
+        return UnDefType.UNDEF;
+    }
+
+    protected State getHumidity() {
+        int unscaledHumidity = bytes[6] & 0xFF;
+        if (unscaledHumidity <= 200) {
+            double scaledHumidity = unscaledHumidity * 0.5;
+            return new DecimalType(scaledHumidity);
+        }
+        return UnDefType.UNDEF;
+    }
+
+    protected State getIllumination() {
+        int illumination = ((bytes[7] & 0xFF) << 8) | (bytes[8] & 0xFF);
+        if (illumination <= 60000) {
+            return new QuantityType<>(illumination, Units.LUX);
+        }
+        return UnDefType.UNDEF;
+    }
+
+    protected State getBatteryLevel() {
+        int unscaledBatteryLevel = ((bytes[9] & 0xFF) >> 3);
+        if (unscaledBatteryLevel <= 20) {
+            return new DecimalType(unscaledBatteryLevel * 5);
+        }
+        return UnDefType.UNDEF;
+    }
+
+    @Override
+    protected String convertToEventImpl(String channelId, String channelTypeId, String lastEvent,
+            Configuration config) {
+
+        // Sensor values
+        if (bytes[0] == 0x00) {
```

Comment:
To support readability I would add an enum (or at least a const) for the different message types with their id. You could replace all the other magic numbers (at least the different states like sash, handle, motion) with a const, too.

Reply:
I added some enums for the message type and the sash, handle and motion states.